src/UIAutomation/FunctionalTests/UiaOperationAbstractionTests.cpp

```diff
@@ -174,6 +174,121 @@ namespace UiaOperationAbstractionTests
+
+            auto scope = UiaOperationScope::StartNew();
+
+            UiaElement element = calc;
+
+            auto defaultVal = element.GetPropertyValue(UiaPropertyId(UIA_AriaPropertiesPropertyId), false);
+            scope.BindResult(defaultVal);
+            auto ariaProperties = defaultVal.AsString();
+            scope.BindResult(ariaProperties);
+
+            auto notSupportedVal = element.GetPropertyValue(UiaPropertyId(UIA_AriaPropertiesPropertyId), true);
+            scope.BindResult(notSupportedVal);
+
+            scope.Resolve();
+
+            Assert::AreEqual(static_cast<bool>(defaultVal.IsNotSupported()), false);
+            Assert::AreEqual(std::wstring(static_cast<wil::shared_bstr>(ariaProperties).get()), std::wstring(L""));
+            Assert::AreEqual(static_cast<bool>(notSupportedVal.IsNotSupported()), true);
+        }
+
+        TEST_METHOD(GetPropertyValueIgnoreDefaultLocalTest)
+                {
+            GetPropertyValueIgnoreDefaultTest(false);
+        }
+
+        TEST_METHOD(GetPropertyValueIgnoreDefaultRemoteTest)
+        {
+            GetPropertyValueIgnoreDefaultTest(true);
+        }
+
+        // Tests GetPropertyValue with useCachedAPI set to true and false
+        // Ensuring that an exception is raised if trying to fetch a cached value that is not cached. 
+        void GetPropertyValueUseCachedAPITest(const bool useRemoteOperations)
+                {
+            ModernApp app(L"Microsoft.WindowsCalculator_8wekyb3d8bbwe!App");
+            app.Activate();
+            auto calc = WaitForElementFocus(L"Display is 0");
+
+            auto guard = InitializeUiaOperationAbstraction(useRemoteOperations);
+
+            auto scope = UiaOperationScope::StartNew();
+
+            UiaElement elementWithoutCache = calc;
+            scope.BindResult(elementWithoutCache);
+
+            UiaOperationAbstraction::UiaCacheRequest cacheRequest;
+            cacheRequest.AddProperty(UIA_NamePropertyId);
+
+            auto elementWithCache = elementWithoutCache.GetUpdatedCacheElement(cacheRequest);
+scope.BindResult(elementWithCache);
+
+            scope.Resolve();
+
+            UiaString cachedName = elementWithCache.GetPropertyValue(UiaPropertyId(UIA_NamePropertyId), /*ignoreDefault=*/ false, /*useCachedApi=*/ true).AsString();
+            Assert::AreEqual(std::wstring(static_cast<wil::shared_bstr>(cachedName).get()), std::wstring(L"Display is 0"));
+
+            Assert::ExpectException<winrt::hresult_error>([&]()
+                {
+                elementWithoutCache.GetPropertyValue(UiaPropertyId(UIA_NamePropertyId), /*ignoreDefault=*/ false, /*useCachedApi=*/ true);
+            });
+
+            UiaString uncachedName = elementWithoutCache.GetPropertyValue(UiaPropertyId(UIA_NamePropertyId), /*ignoreDefault=*/ false, /*useCachedApi=*/ false).AsString();
+            Assert::AreEqual(std::wstring(static_cast<wil::shared_bstr>(uncachedName).get()), std::wstring(L"Display is 0"));
```

Comment:
For completeness, validate that we can use `useCachedApi = false` with `elementWithCache` as well.